share/modules/cmake/CMTest.cmake

```diff
@@ -51,11 +51,11 @@ function(cm_mark_as_test)
 endfunction(cm_mark_as_test)
 
 
-function(cm_create_internal_targets)
-    if (NOT TARGET _cm_internal_tests-${CMAKE_WORKSPACE_NAME}-${PROJECT_NAME})
-        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/_cm_internal_tests-${CMAKE_WORKSPACE_NAME}-${PROJECT_NAME}.cpp "")
-        add_library(_cm_internal_tests-${CMAKE_WORKSPACE_NAME}-${PROJECT_NAME} STATIC ${CMAKE_CURRENT_BINARY_DIR}/_cm_internal_tests-${CMAKE_WORKSPACE_NAME}-${PROJECT_NAME}.cpp)
-        cm_mark_as_test(_cm_internal_tests-${CMAKE_WORKSPACE_NAME}-${PROJECT_NAME})
+function(cm_create_internal_targets NAME)
+    if (NOT TARGET ${internal-lib-name})
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${internal-lib-name}.cpp "")
+        add_library(${internal-lib-name} STATIC ${CMAKE_CURRENT_BINARY_DIR}/${internal-lib-name}.cpp)
+        cm_mark_as_test(${internal-lib-name})
```

Comment:
`NAME` instead of `${internal-lib-name}`. Now it is using value from parent scope I guess